### PR TITLE
[cmake] Initial integration

### PR DESCRIPTION
**Pull request by AdamKorcz**
Fuzzer being uploaded upstrem here: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5668

### CODE REVIEW

**Comment by AdamKorcz** [comment]
@jonathanmetzman @inferno-chromium Ready for review.

@bradking - Letting you know for info.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: projects/cmake/build.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+################################################################################
+
+# Build CMake
+mkdir build-dir && cd build-dir
+../bootstrap && make -j$(nproc)
+
+
+# Build fuzzers
+cd ../Tests/Fuzzing
+$CXX $CXXFLAGS 	-I../../Source \
```
Comment (jonathanmetzman) [inline projects/cmake/build.sh:25]: nit: can you fix this spacing.

File: projects/cmake/build.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		-I../../build-dir/Source \
+		-c xml_parser_fuzzer.cc \
+		-o xml_parser_fuzzer.o 
+
+
+export cmexpat_dir="${SRC}/CMake/build-dir/Utilities/cmexpat/CMakeFiles/cmexpat.dir/lib"
+$CXX $CXXFLAGS $LIB_FUZZING_ENGINE \
+		xml_parser_fuzzer.o -o $OUT/xml_parser_fuzzer \
```
Comment (jonathanmetzman) [inline projects/cmake/build.sh:33]: nit: fix this spacing please.

File: projects/cmake/build.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Build CMake
```
Comment (jonathanmetzman) [inline projects/cmake/build.sh:18]: nit: end comments with periods.

**Comment by AdamKorcz** [comment]
Sorry for the delay and thank you for the fixes!

**Pull request merged**